java-checks/src/main/java/org/sonar/java/checks/CommentedOutCodeLineCheck.java

```diff
@@ -53,6 +58,15 @@ public List<Tree.Kind> nodesToVisit() {
     return Collections.singletonList(Tree.Kind.TOKEN);
   }
 
+  @Override
+  public void setContext(JavaFileScannerContext context) {
+    super.setContext(context);
+    compilationUnitFirstTokenPosition = Optional.ofNullable(context.getTree())
+      .map(Tree::firstToken)
+      .map(SyntaxToken::range)
+      .map(Range::start).orElse(FILE_START);
+  }
+
```

Comment:
Would it make sense to reset `compilationUnitFirstTokenPosition` to `FILE_START` when leaving the file?
```suggestion

  @Override
  public void leaveFile(JavaFileScannerContext context) {
    compilationUnitFirstTokenPosition = FILE_START;
  }

```

Reply:
I thought about it and decided that there's not enough benefit to justify adding some code. Because the Position class is just a small class with two `int` fields. There's not risk to keep the full AST in memory in case of exception. And there's no risk to use an invalid `compilationUnitFirstTokenPosition` form the previous AST because of the `setContext` execution.

Conclusion: Yes we can but I'm not sure we should.